content/metastore-integrations/docs/hive-metastore.md

```diff
@@ -151,4 +157,213 @@ aws emr-containers start-job-run --cli-input-json file:///Spark-Python-in-s3-hms
+
+helm repo add hive-metastore https://melodyyangaws.github.io/hive-metastore-chart 
+helm install hive hive-metastore/hive-metastore -f values.yaml --namespace=emr --debug
+```
+
+**hivethrift_eks.py**
+
+```python
+from os import environ
+import sys
+from pyspark.sql import SparkSession
+
+spark = SparkSession \
+    .builder \
+    .config("spark.sql.warehouse.dir",environ['warehouse_location']) \
+    .config("hive.metastore.uris","thrift://"+environ['HIVE_METASTORE_SERVICE_HOST']+":9083") \
+    .enableHiveSupport() \
+    .getOrCreate()
+
+spark.sql("SHOW DATABASES").show()
+spark.sql("CREATE DATABASE IF NOT EXISTS `demo`")
+spark.sql("DROP TABLE IF EXISTS demo.amazonreview3")
+spark.sql("CREATE EXTERNAL TABLE IF NOT EXISTS `demo`.`amazonreview3`( `marketplace` string,`customer_id`string,`review_id` string,`product_id` string,`product_parent` string,`product_title` string,`star_rating` integer,`helpful_votes` integer,`total_votes` integer,`vine` string,`verified_purchase` string,`review_headline` string,`review_body` string,`review_date` date,`year` integer) STORED AS PARQUET LOCATION '"+sys.argv[1]+"/app_code/data/toy/'")
+
+``` 
+An environment variable `HIVE_METASTORE_SERVICE_HOST` appears in your Spark application pods automatically, once the standalone HMS is up running in EKS. You can directly set the `hive.metastore.uris` to `thrift://"+environ['HIVE_METASTORE_SERVICE_HOST']+":9083"`.
+
+Can set the `spark.sql.warehouse.dir` property to a S3 location as your hive warehouse storage. The s3 location can be dynamic, which is based on an argument passed in or an environment variable.
+
+**Request:**
+
+```bash
+#!/bin/bash
+aws emr-containers start-job-run \
+--virtual-cluster-id $VIRTUAL_CLUSTER_ID \
+--name spark-hive-via-thrift \
+--execution-role-arn $EMR_ROLE_ARN \
+--release-label emr-6.2.0-latest \
+--job-driver '{
+  "sparkSubmitJobDriver": {
+      "entryPoint": "s3://'$S3BUCKET'/app_code/job/hivethrift_eks.py",
+      "entryPointArguments":["s3://'$S3BUCKET'"],
+      "sparkSubmitParameters": "--conf spark.driver.cores=1 --conf spark.executor.memory=4G --conf spark.driver.memory=1G --conf spark.executor.cores=2"}}' \
+--configuration-overrides '{
+    "monitoringConfiguration": {
+      "s3MonitoringConfiguration": {"logUri": "s3://'$S3BUCKET'/elasticmapreduce/emr-containers"}}}'
+```
+      
+### **4-Run thrift service as a sidecar in Spark Driver's pod**
+
+This advanced solution runs the standalone HMS thrift service inside a Spark driver as a sidecar. It means each Spark job will have its dedicated thrift server. The benefit of the design is HMS is no long a single point of failure, since each Spark application has its own HMS. Also it is no long a long running service, i.e. it spins up when your Spark job starts, then terminates when your job is done. The disadvantage is the initial setup against an existing EKS cluster is not easy and it requires combination skills of analytics and k8s. 
```

Comment:
I'm not sure if I agree with disadvantage since you are already providing the solution. You may want to say maintenance of container image increases because you now have to maintain hms sidecar. Also you will need combination of analytics and k8s skill to maintain this solution

Reply:
Does the sidecar uses secrets from Secrets Manager? I don't remember if it did. If not, I'd add that as a con bcoz k8s secrets are base64 encoded and human readable. Customers may want to secure access to k8s secrets. Also, if they need secure solution, review your other solution which has integration with Secrets Manager